add support for SAMPLE-AES decryption

erankor opened the pull request:
### Description of the Changes
Add support for decryption of SAMPLE-AES for AVC and AAC codecs

### CheckLists

- [V] changes have been done against master branch, and PR does not conflict
- [V] no commits have been done in dist folder (we will take care of updating it)
- [X] new unit / functional tests have been added (whenever applicable)
- [V] Travis tests are passing (or test results are not worse than on master branch :))
- [V] API or design changes are documented in API.md

mangui commented:
awesome @erankor !
do you have test streams in hands ?

erankor commented:
I'm generating streams with our packager (https://github.com/kaltura/nginx-vod-module/) here's a sample URL: http://lbd.kaltura.com:8001/local/ehls/content/bbb.mp4/index.m3u8
(it's on a dev server, so may go down at some point, but for now it's there)

mangui commented:
FYI I mirrored your stream on http://streambox.fr/playlists/sample_aes/index.m3u8
could you add it in streams.json so that AES sample decryption could be automatically checked on every commit ?

mangui commented:
it works great, however the debug logs are a bit too verbose as we got a debug logs for each audio/video samples.
![image](https://cloud.githubusercontent.com/assets/1798553/23214643/c5b084c8-f90f-11e6-83a6-f890b4551012.png)

IMHO it should be a once-off log.
it would be great to find a way to switch it to a once-off log.

erankor commented:
@mangui 
1. added the stream to streams.json
2. added a `logEnabled` boolean on Decrypter to reduce the amount of logging - it will print 1-3 lines per segment, I believe that's how it was before these changes (3 lines are printed in case of fallback from WebCrypto to JS)
3. resolved conflicts following latest updates to master

Thanks!

mangui commented:
perfect, as you can see I just merged and added https://github.com/dailymotion/hls.js/commit/9ae9b4ab301922eb1d350808cbd3ee712a32fd7d that should fix tests failing on Travis/Saucelabs

mangui commented:
🍾 https://travis-ci.org/dailymotion/hls.js/builds/204524086 🍾
